### DISCUSSION TITLE
Why are SQL indexes not supported?

### DISCUSSION THREAD

**Discussion by wy0wxb** [ndsev/zserio#441] [Q&A] [answered]
I want to set the multi-field index, but I can't find the instructions

**Comment by mikir** [comment]
Zserio language does not support SQLite indexes at all, so unfortunately this is not possible.

**Comment by mkutscherauer** [comment]
Shouldn't this work by using the sql keyword similiar to how the primary key is specified here:

https://github.com/ndsev/zserio/blob/master/doc/ZserioLanguageOverview.md#sqlite-tables

**Comment by mikir** [comment]
Yes, if there is a need to support SQLite indexes, we will probably implement it as you wrote. We just did not get any request to support SQLite indexes so far, so we have decided not to implement them to save effort.

**Comment by fklebert** [comment, marked as answer]
A note from past experience: Setting a SQL index during schema generation time is in almost all cases not optimal, as it heavily depends on the filling of a table later on how you can optimze performance by adding indexes.
That is why we have not implemented them in zserio so far.
